dev/scripts/src/contract/import.ts

```diff
@@ -11,12 +11,77 @@
+import { exec } from '../util/index.js'
+import fs from 'fs'
+import path from 'path'
 
-async function importContract(relPathToABIs: string, relPathToOutput: string): Promise<ExecOutput> {
+async function importContract(pathToAbis: string, pathToOutput: string) {
+    pathToAbis = path.relative(process.cwd(), pathToAbis)
+    pathToOutput = path.relative(process.cwd(), pathToOutput)
     //TODO import typechain when it's working https://github.com/Brushfam/typechain-polkadot/issues/73
-    const cmd = `npx @727-ventures/typechain-polkadot --in ${relPathToABIs} --out ${relPathToOutput}`
-    return exec(cmd)
+    if (!fs.existsSync(pathToAbis)) throw new Error(`Path to ABIs does not exist: ${pathToAbis}`)
+    await exec(`mkdir -p ${pathToOutput}`)
+    const cmd = `npx @727-ventures/typechain-polkadot --in ${pathToAbis} --out ${pathToOutput}`
+    await exec(cmd)
+    // walk each file in the output directory
+    const walk = (dir: string) => {
+        const files = fs.readdirSync(dir)
+        for (const file of files) {
+            const filePath = path.join(dir, file)
+            const stat = fs.statSync(filePath)
+            if (stat.isDirectory()) {
+                walk(filePath)
+            } else if (filePath.endsWith('.ts')) {
+                // replace the relative imports with .js extension applied
+                // eslint-disable-next-line no-useless-escape
+                const regex = /import[\s]+(.+?)[\s]+from[\s]+([\"\']\..+)/gm
+                const fileContents = fs.readFileSync(filePath, 'utf8')
+                let replaced = fileContents.replace(regex, (match, p1, p2) => {
+                    const name = p1.toString()
+                    const srcQuoted = p2.toString()
+                    const src = getPath(srcQuoted)
+                    const extension = getExtension(src)
+                    let result = ''
+                    if (extension === 'js') {
+                        // already has .js extension
+                        return match
+                    } else if (extension === 'json') {
+                        // needs json assertion
+                        result = `import ${name} from '${src}' assert { type: 'json' }`
+                    } else {
+                        // needs .js extension
+                        result = `import ${name} from '${src}.js'`
+                    }
+                    console.log(`Replacing \n\t${match}\nwith\n\t${result}\nin ${filePath}`)
+                    return `${result}`
+                })
+                // eslint-disable-next-line no-useless-escape
+                replaced = replaced.replace(/(^(?!\s*\/\/\s*eslint).*)\n+(\s*)\/\/\s*@ts-ignore/gm, (match, p1, p2) => {
+                    const result = `${p1}\n${p2}// eslint-disable-next-line @typescript-eslint/ban-ts-comment\n${p2}// @ts-ignore`
+                    console.log(`Replacing \n\t${match}\nwith\n\t${result}\nin ${filePath}`)
+                    return result
```

Comment:
Can we refactor this file when next working in this area? Would benefit from breaking up into helper functions, strings to named constants

Also could be using the logger from common if that's not gonna cause deps issues

Reply:
we can, but all of this should be temporary because ideally typechain will fix everything here on there end, then we don't have to do any post-processing